File: Domain/Models/Fields/GuardianField.cs
﻿using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class GuardianField : BaseModel
    {
        public int GuardianFieldId { get; set; }

        [Required]
        [MaxLength(100)]
        public string Value { get; set; }

        [Required]
        public int FieldId { get; set; }
        public virtual Field Field { get; set; }

        [Required]
        public int GuardianId { get; set; }
    }
}

File: Client/lib/models/Field/GuardianField.dart
import 'package:Client/models/Base/BaseModel.dart';
import 'package:Client/models/Field/FieldModel.dart';

class GuardianFieldModel extends BaseModel {
  GuardianFieldModel(
      {int id, this.field, this.fieldId, this.value, this.guardianTypeId})
      : super(id: id);
  String value;
  int guardianId;
  int fieldId;
  int guardianTypeId;
  FieldModel field;

  @override
  GuardianFieldModel.fromJson(Map<String, dynamic> map) : super.fromJson(map) {
    id = toInt(map['phoneId']);
    guardianId = toInt(map['guardianId']);
    fieldId = toInt(map['fieldId']);
    guardianTypeId = toInt(map['guardianTypeId']);
    value = map['value'];
    field = map['field'] == null ? null : FieldModel.fromJson(map['field']);
  }

  @override
  Map<String, dynamic> toJson() {
    var map = super.toJson();
    map.addAll({
      "guardianFieldId": id,
      "guardianId": guardianId,
      "fieldId": fieldId,
      "guardianTypeId": guardianTypeId,
      "value": value,
      "field": field?.toJson(),
    });
    return map;
  }
}
